File: java/JavaFileutil/src/JavaFileutil/Validators/DirectoryNonExistenceValidator.java
package JavaFileutil.Validators;

import java.util.List;

public class DirectoryNonExistenceValidator extends AbstractFilesystemEntryValidator
{
    public DirectoryNonExistenceValidator(List<String> args, short index, String message)
    {
        super(args, index, message);
    }

    @Override
    public boolean Validate()
    {
        if (file.exists() && file.isDirectory())
        {
            System.out.println(ErrorPreface + Message);
            return false;
        }

        return true;
    }
}

File: python/Validators/DirectoryNonExistenceValidator.py
from Abstractions.AbstractFilesystemEntryValidator import *


class DirectoryNonExistenceValidator(AbstractFilesystemEntryValidator):
    def __init__(self, validate_this_dir, message):
        super().__init__(validate_this_dir, message)

    def validate(self):
        if self._validate_this.exists() and self._validate_this.is_dir():
            self._print_validation_error()
            return False
        return True
